src/app/my-absences/components/my-absences-confirm/my-absences-confirm.component.ts

```diff
@@ -21,7 +21,13 @@ import { ConfirmAbsencesSelectionService } from 'src/app/shared/services/confirm
 })
 export class MyAbsencesConfirmComponent extends MyAbsencesAbstractConfirmComponent {
   selectedLessonIds$ = this.selectionService.selectedIds$.pipe(
-    map((selectedIds) => selectedIds[0]?.lessonIds || [])
+    map((selectedIds) =>
+      selectedIds
+        .map((s) => s.lessonIds)
+        .reduce((acc, lessonIds) => {
+          return acc.concat(lessonIds);
+        }, [])
+    )
```

Comment:
Kleine Kosmetik:

Du loopst mit dem `map` und dem `reduce` zweimal über alle Einträge. Ich würde das in diesem Fall einfach in einem Schritt machen:

```
    map((selectedIds) =>
      selectedIds
        .reduce((acc, s) => [...acc, ...s.lessonIds], [])
    )
```

Eine Alternative wäre [flatten](https://lodash.com/docs/4.17.15#flatten) zu verwenden:

```
    map((selectedIds) => uniq(flatten(selectedIds.map(s => s.lessonIds))))
```

Sollte nicht zusätzlich auch ein [uniq](https://lodash.com/docs/4.17.15#uniq) gemacht werden, damit diese Lektionen nicht mehrfach vorkommen?

Reply:
Danke. Ich habe das nun so  umgesetzt (für alle drei Fälle).